plugins/processors/t128_filter/t128_filter_test.go

```diff
@@ -0,0 +1,120 @@
+package t128_filter
+
+import (
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/influxdata/toml"
+	"github.com/stretchr/testify/assert"
+)
+
+var t0 = time.Unix(0, 0)
+
+func newMetric(name string, tags map[string]string, fields map[string]interface{}) telegraf.Metric {
+	if tags == nil {
+		tags = map[string]string{}
+	}
+	if fields == nil {
+		fields = map[string]interface{}{}
+	}
+	m, _ := metric.New(name, tags, fields, t0)
+	return m
+}
+
+func TestFilters(t *testing.T) {
+	testCases := []struct {
+		Name          string
+		Conditions    []Condition
+		InputMetrics  []telegraf.Metric
+		OutputMetrics []telegraf.Metric
+	}{
+		{
+			Name:          "passes through with no conditions",
+			Conditions:    []Condition{},
+			InputMetrics:  []telegraf.Metric{newMetric("some-measurement", nil, nil)},
+			OutputMetrics: []telegraf.Metric{newMetric("some-measurement", nil, nil)},
+		},
+		{
+			Name:       "drops",
+			Conditions: []Condition{{Tags: tags{"tag1": {"value1"}}}},
+			InputMetrics: []telegraf.Metric{
+				newMetric("some-measurement", map[string]string{"tag1": "value1"}, nil),
+				newMetric("some-measurement", map[string]string{"tag1": "value2"}, nil),
+			},
+			OutputMetrics: []telegraf.Metric{newMetric("some-measurement", map[string]string{"tag1": "value1"}, nil)},
+		},
+		{
+			Name:       "drops if no tag",
+			Conditions: []Condition{{Tags: tags{"tag1": {"value1"}}}},
+			InputMetrics: []telegraf.Metric{
+				newMetric("some-measurement", nil, nil),
+			},
+			OutputMetrics: []telegraf.Metric{},
+		},
+		{
+			Name:       "ands conditions together",
```

Comment:
I dont think this is right. The conditions need to be OR'd together not AND'd by default.